continuationsPlugin/src/main/scala/continuations/DefDefTransforms.scala

```diff
@@ -128,69 +142,76 @@ class DefDefTransforms(using Context) {
+        case CallsContinuationResumeWith(resumeArg) =>
+          transformSuspendNoParametersOneContinuationResume(tree, resumeArg)
+        case _ =>
+          transformSuspendedDef(tree, cpy)
+    report.logWith(s"new tree:")(newTree)
+
+  def transformSuspendContinuation(tree: tpd.DefDef)(using Context): tpd.DefDef =
+    tree match
+      case ReturnsContextFunctionWithSuspendType(_) => transformContinuationWithSuspend(tree)
+      case HasSuspendParameter(_) => transformContinuationWithSuspend(tree)
+      case _ => report.logWith(s"oldTree:")(tree)
 
-  def transformSuspendContinuation(tree: DefDef): DefDef =
+  private def transformNonSuspending(tree: tpd.DefDef, cpy: TypedTreeCopier)(
+      using ctx: Context): tpd.DefDef =
+    tree match
+      case Trees.DefDef(name, paramss, tpt, rhs) =>
+        val suspendClazz = requiredClass(suspendFullName)
+        val completion: tpd.ValDef = tpd.ValDef(
+          newSymbol(
+            tree.symbol,
+            Names.termName("completion"),
+            Flags.LocalParam,
+            requiredPackage("continuations")
+              .requiredType("Continuation")
+              .typeRef
+              .appliedTo(Types.OrType(tpt.tpe, ctx.definitions.AnyType, false))
+          ).asTerm,
+          rhs = theEmptyTree
+        )
+        val params: List[tpd.ParamClause] = if (paramss.isEmpty) {
+          List(List(completion))
+        } else
+          tree
+            .paramss
+            .zipWithIndex
+            .map { (pc, i) =>
+              val newPc: tpd.ParamClause = pc
+                .filterNot { param =>
+                  param match {
+                    case p: Trees.ValDef[Type] =>
+                      p.typeOpt.hasClassSymbol(suspendClazz) && p
+                        .symbol
+                        .flags
+                        .isOneOf(Flags.GivenOrImplicit)
+                    case t: Trees.TypeDef[Type] =>
+                      t.typeOpt.hasClassSymbol(suspendClazz) && t
+                        .symbol
+                        .flags
+                        .isOneOf(Flags.GivenOrImplicit)
+                  }
+                }
+                .asInstanceOf[tpd.ParamClause]
+              if (i == 0) {
+                newPc.appended(completion).asInstanceOf[tpd.ParamClause]
+              } else newPc
+            }
+            .filterNot(_.isEmpty)
+
+        cpy.DefDef(tree)(
+          paramss = params,
+          tpt = tpd.TypeTree(ctx.definitions.ObjectClass.typeRef))
+
+  def transformSuspendedDef(tree: tpd.DefDef, cpy: TypedTreeCopier)(
+      using ctx: Context): tpd.DefDef =
     tree match
-      case HasSuspendParameter(_) =>
-        tree match
-          case CallsContinuationResumeWith(resumeArg) =>
-            transformSuspendNoParametersOneContinuationResume(tree, resumeArg)
-          case _ => tree
+      case BodyHasSuspensionPoint(_) => cpy.DefDef(tree)()
+      case HasSuspendParameter(_) => transformNonSuspending(tree, cpy)
```

Comment:
This path has already been checked in `transformSuspendContinuation`.

Reply:
Addressed in https://github.com/47deg/TBD/pull/53/commits/3cbb21aaf70208a1db408fa9e4cb5eda55952e34

Reply:
> I don't see that IsSuspendedType, IsCallToSuspend and HasInnerSuspensionPoint classes are being used here, should we remove them?

Missed this. I'll clear them out in the morning.